regions/regions.go

```diff
@@ -44,13 +45,46 @@ func (rg Regions) RoundTrip(req *http.Request) (*http.Response, error) {
 	}
 	shardsRing, ok := rg.multiCluters[reqHost]
 	if ok {
+		req.Header.Add(storage.HOST, reqHost)
+		req.Header.Add(storage.BUCKET, "")
+		return shardsRing.DoRequest(req)
+	}
+	host, bucketName, hostFound := rg.findHostInDomainStyle(reqHost)
+	if hostFound {
+		shardsRing, _ = rg.multiCluters[host]
+		req.Header.Add(storage.HOST, host)
+		req.Header.Add(storage.BUCKET, bucketName)
 		return shardsRing.DoRequest(req)
+
 	}
 	if rg.defaultRing != nil {
 		return rg.defaultRing.DoRequest(req)
 	}
 	return rg.getNoSuchDomainResponse(req), nil
 }
+func (rg Regions) findHostInDomainStyle(originalHost string) (currentHost, bucket string, ok bool) {
+
+	currentHost = ""
+	lastSubDomainIndex := strings.LastIndex(originalHost, ".")
+
+	for lastSubDomainIndex != -1 {
+
+		currentHost = originalHost[lastSubDomainIndex+ 1:] + currentHost
+
+		_, ok = rg.multiCluters[currentHost]
+
+		if ok {
+			return currentHost, originalHost[:lastSubDomainIndex], ok
```

Comment:
What if I told you that you may have two different rings one have assigned akubra.net and other other.akubra.net?

Reply:
I'd say "damn, you're right"

Reply:
Fixed